src/CronExpressionParser.ts

```diff
@@ -0,0 +1,273 @@
+   * @private
+   * @returns {(number | string)[]} The parsed field.
+   */
+  static #parseField(field: keyof typeof CronUnit, value: string, constraints: CronConstraints): (number | string)[] {
+    // Replace aliases for month and dayOfWeek
+    if (field === 'Month' || field === 'DayOfWeek') {
+      value = value.replace(/[a-z]{3}/gi, (match) => {
+        match = match.toLowerCase();
+        const replacer = Months[ match as keyof typeof Months ] || DayOfWeek[ match as keyof typeof DayOfWeek ];
+        assert(replacer != null, `Validation error, cannot resolve alias "${match}"`);
+        return replacer.toString();
+      });
+    }
+
+    // Check for valid characters
+    assert(constraints.validChars.test(value), `Invalid characters, got value: ${value}`);
+
+    // Replace '*' and '?'
+    value = value.replace(/[*?]/g, constraints.min + '-' + constraints.max);
+    return CronExpressionParser.#parseSequence(value, constraints, field);
+  }
+
+  /**
+   * Parse a sequence from a cron expression.
+   * @param {string} val - The sequence to parse.
+   * @param {CronConstraints} constraints - The constraints for the field.
+   * @param {keyof typeof CronUnit} field - The field to parse.
+   * @private
+   */
+  static #parseSequence(val: string, constraints: CronConstraints, field: keyof typeof CronUnit): (number | string)[] {
+    const stack: (number | string)[] = [];
+
+    function handleResult(result: number | string | (number | string)[], constraints: CronConstraints) {
+      if (Array.isArray(result)) {
+        result.forEach((value) => {
+          /* istanbul ignore else - FIXME no idea how this is triggered or what it's purpose is */
+          if (!CronExpressionParser.#isValidConstraintChar(constraints, value)) {
+            const v = parseInt(value.toString(), 10);
+            const isValid = v >= constraints.min && v <= constraints.max;
+            assert(isValid, `Constraint error, got value ${value} expected range ${constraints.min}-${constraints.max}`);
+            stack.push(value);
+          } else {
+            /* istanbul ignore next - FIXME no idea how this is triggered or what it's purpose is */
+            stack.push(value);
+          }
+        });
+      } else {
+        if (CronExpressionParser.#isValidConstraintChar(constraints, result)) {
+          stack.push(result);
+        } else {
+          const v = parseInt(result.toString(), 10);
+          const isValid = v >= constraints.min && v <= constraints.max;
+          assert(isValid, `Constraint error, got value ${result} expected range ${constraints.min}-${constraints.max}`);
+          stack.push(field === 'DayOfWeek' ? v % 7 : result);
+        }
+      }
+    }
+
+    const atoms = val.split(',');
+    assert(atoms.every((atom) => atom.length > 0), 'Invalid list value format');
+    atoms.forEach((atom) => handleResult(CronExpressionParser.#parseRepeat(atom, constraints, field), constraints));
```

Comment:
```suggestion
    atoms.forEach((atom) => {
      assert(atom.length > 0, 'Invalid list value format');
      handleResult(CronExpressionParser.#parseRepeat(atom, constraints, field), constraints);
    });
```

Not that `atoms.every` causes that much overhead but probably won't need extra loop here to verify `atoms` contents.